pkg/deployment/agent.go

```diff
@@ -93,8 +93,12 @@ func (a *Agent) Get() *appsv1.DaemonSet {
 	sort.Strings(args)
 
 	hostNetwork := false
+	dnsPolicy := a.jaeger.Spec.Agent.DNSPolicy
 	if a.jaeger.Spec.Agent.HostNetwork != nil {
 		hostNetwork = *a.jaeger.Spec.Agent.HostNetwork
+		if dnsPolicy == "" {
+			dnsPolicy = corev1.DNSClusterFirstWithHostNet
```

Comment:
Do you need to set a default value here? Is that different than if a value simply isn't provided?

Reply:
According to the [k8s documentation](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy),  if we do not set `dnsPolicy` to `DNSClusterFirstWithHostNet`, k8s will use `ClusterFirst` by default, and the daemonset agent will not be able to resolve the service ip of the collector properly.
```
Error while dialing dial tcp: lookup jaeger-collector.observability.svc.cluster.local on {hostIP}:53: no such host
```

Reply:
Does it mean that DaemonSets aren't working at the moment? I think we have e2e tests exercising this.

Reply:
Yes, DaemonSets should not be working now. My company has bypassed this problem by forwarding some dns domains to coredns on the physical machine.
I can try adding the corresponding test case to the e2e test.

Reply:
@kevinearls, we do have this, don't we?

@faceair if you could provide an e2e to prove, that would be awesome!

Reply:
I modified the existing es2 test case, is this OK?

Reply:
ping @kevinearls